### PR TITLE
Skip sub-folders and make pattern matching more useful

### PR DESCRIPTION
**Pull request by sgillies**
These changes help unlock 1-click ingestion of folders of SOMA sources.

Unneeded log messages have been removed.

### CODE REVIEW

File: src/tiledb/cloud/soma/ingest.py [hunk 1]
```diff
@@ -168,8 +175,12 @@ def run_ingest_workflow_udf(
             entry_output_uri += base
             logger.debug("ENUMERATOR ENTRY_OUTPUT_URI=%r", entry_output_uri)
 
-            if pattern is not None and not re.match(pattern, entry_input_uri):
-                logger.debug("ENUMERATOR SKIP NO MATCH ON <<%r>>", pattern)
+            if pattern is not None and not re.search(pattern, entry_input_uri):
```
Comment (sgillies) [inline src/tiledb/cloud/soma/ingest.py:178]: `re.search()` matches anywhere in the string.

File: src/tiledb/cloud/soma/ingest.py [hunk 2]
```diff
@@ -168,8 +175,12 @@ def run_ingest_workflow_udf(
...
+                logger.info(
+                    "Skipping non-matching input: pattern=%r, entry_input_uri=%r",
```
Comment (sgillies) [inline src/tiledb/cloud/soma/ingest.py:180]: I'm going to start nudging all log entry into this form. `Event happened: var1=foo, var2=bar`.

**Comment by Tile-Kyle** [comment]
can I confirm the changes here will skip any sub folders from the `source` input path? which I believe was included in the original [PRD](https://www.notion.so/SOMA-1-Click-Ingestion-95962a7e579144fcbd55af7180b87b93?pvs=4#e48b1eb08256464dbbc5e6b94e611265) - but just checking. Since we will need to communicate this to the user on the UI. Thanks!

File: src/tiledb/cloud/soma/ingest.py [hunk 3]
```diff
@@ -103,44 +103,13 @@ def run_ingest_workflow_udf(
...
-            resources=carry_along.get("resources", resources),
-            access_credentials_name=carry_along.get("access_credentials_name", acn),
-            logging_level=logging_level,
-            dry_run=dry_run,
-        )
-
-    elif vfs.is_dir(input_uri):
-        logger.debug("ENUMERATOR VFS.IS_DIR")
-
+    if vfs.is_dir(input_uri):
```
Comment (sgillies) [inline src/tiledb/cloud/soma/ingest.py:112]: Here's the fix! `vfs.is_file(input_uri)` returns `True`, erroneously. We need to always test with `is_file()` first.

**Comment by sgillies** [comment]
@Tile-Kyle yes, that's right. Only top-level contents of a folder are to be ingested.

**Comment by sgillies** [comment]
@ktsitsi tagging you for review because I want you to see this issue with `VFS`.

**Review by johnkerl — APPROVED** [review]
🚢

**Pull request merged**